clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1309,7 +1362,8 @@ private void processRequests() {
          * @param now       The current time in milliseconds.
          */
         void enqueue(Call call, long now) {
-            if (call.tries > maxRetries) {
```

Comment:
As your comment suggests, I think a cleaner way to have this code would be to use `enqueue` as specifically for adding additional calls and no retrying logic. All retrying logic should be consolidated to the new retry context and call class, so that there's just one place we're checking for everything. This means we have to ensure that before we call `enqueue` or any other method that calls this, we are doing the correct checks to ensure that we are checking that we are able to retry.

Reply:
Refactored. Now Call::call is the only method interacting with Call::enqueue. Other Call::enqueue usages are replaced by retry. Now the retry would also judge if the retry backoff is applicable to the exception thrown by failure.

Reply:
Seems like we still need to judge this condition. For example, the find coordinator call will asynchronously "call()" the "nextCalls" after receiving the response. Although it's a retry, it can't use our "retry()" and the number of retries won't get checked. Hence, I added the check back to the "call()" method.

Reply:
Since we already got the refactoring, the above comment seems unnecessary if we make `enqueue` private.